src/figuresClasses.ts

```diff
@@ -1,19 +1,88 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape?: Shape,
+  color: Color,
+  getArea(): number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sides
+      = Object.values(this)
+        .filter((value) => typeof value === 'number')
+        .sort((x, y) => x - y);
 
-export class Circle {
+    if (
+      Object.values(this).some((value) => typeof value === 'number'
+      && value <= 0)
+      || sides[0] + sides[1] <= sides[2]
+    ) {
+      throw new Error('Invalid sides values');
```

Comment:
It would be great to specify why exactly the sides are invalid
![image](https://github.com/mate-academy/ts_figures_classes/assets/96409563/cde98986-d881-4b6b-8313-7c81ed1f26b8)